Card.java

```diff
@@ -1,16 +1,37 @@
+import java.util.Comparator;
+
+class CardComparator implements Comparator<Card> {
+
+  public int compare(Card c1, Card c2) {
+    if (c1.getSuit().equals(c2.getSuit())) {
+      return new RankComparator().compare(c1.getRank(), c2.getRank());
+    } else {
+      return new SuitComparator().compare(c1.getSuit(), c2.getSuit());
+    }
+  }
+}
+
 public class Card {
 
-  private int rank;
+  private Rank rank;
   private Suit suit;
 
-  public Card(int rank, Suit suit) {
+  public Rank getRank() {
+    return rank;
+  }
+
+  public Suit getSuit() {
+    return suit;
+  }
+
+  public Card(Rank rank, Suit suit) {
     this.suit = suit;
     this.rank = rank;
   }
 
   @Override
   public String toString() {
-    return rank + " of " + suit;
+    return rank.toString() + suit.toString();
   }
 
   @Override
```

Comment:
There is an extremely important thing you need to do:

- Whenever you override `equals()`, it is strongly encouraged that you also override `hashCode()` so that it remains conformant to its [contract](https://docs.oracle.com/javase/10/docs/api/java/lang/Object.html#hashCode()).

There is a very handy utility class in the Standard Java Library, `java.util.Objects`, which offers the method [hash](https://docs.oracle.com/javase/10/docs/api/java/util/Objects.html#hash(java.lang.Object...)), that you may find very useful in implementing this method.